src/html-splitter.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import * as osPath from 'path';
+import * as logging from 'plylog';
+import {Transform} from 'stream';
+import File = require('vinyl');
+import {FileCB} from './streams';
+
+const logger = logging.getLogger('polymer-project');
+const pred = dom5.predicates;
+
+const extensionsForType: {[mimetype: string]: string} = {
+  'text/ecmascript-6': 'js',
+  'application/javascript': 'js',
+  'text/javascript': 'js',
+  'application/x-typescript': 'ts',
+  'text/x-typescript': 'ts',
+};
+
+/**
+ * HTMLSplitter represents the shared state of files as they are passed through
```

Comment:
Code comments are for our developers, not our consumers, right? This is explained for the user in the README.